### PR TITLE
Tensor<1,dim,Number> - let the cross_product functions return its result

### PR DESCRIPTION
**Pull request by tamiko**
Unfortunately, this is an incompatbile change because the old signature
```
void cross_product(Tensor<1, dim> &dst, const Tensor<1, dim> &src);
```
cannot be kept. It clashes with the new signature
```
Tensor<1, dim>
cross_product(const Tensor<1, dim> &src1, const Tensor<1, dim> &src2);
```

Nevertheless, I consider this a very important change. All other function and
operators just return the result. The cross product should do the same.

### CODE REVIEW

**Comment by tamiko** [comment]
In reference to #1583

File: include/deal.II/base/tensor.h [hunk 1]
```diff
@@ -2167,51 +2167,57 @@ Number determinant (const Tensor<1,1,Number> &t)
...
  */
 template <int dim, typename Number>
 inline
-void
-cross_product (Tensor<1,dim,Number>       &dst,
-               const Tensor<1,dim,Number> &src)
+Tensor<1,dim,Number>
+cross_product (const Tensor<1,dim,Number> &src)
 {
   Assert (dim==2, ExcInternalError());
```
Comment (bangerth) [inline include/deal.II/base/tensor.h]: This function only makes sense in dim==2. Can you try what happens if you only provide it for this particular template argument? (I imagine it was written this way to allow for generic code.)

File: source/base/geometry_info.cc [hunk 2]
```diff
@@ -1820,8 +1820,7 @@ namespace internal
     Tensor<1,3>
     wedge_product (const Tensor<1,3> (&derivative)[2])
     {
-      Tensor<1,3> result;
-      cross_product (result, derivative[0], derivative[1]);
+      Tensor<1,3> result = cross_product (derivative[0], derivative[1]);
 
       return result;
```
Comment (bangerth) [inline source/base/geometry_info.cc]: might as well return the result right away here, without the local variable

File: source/fe/mapping_fe_field.cc [hunk 3]
```diff
@@ -1043,10 +1043,11 @@ namespace internal
                                                       -1 : +1);
                   break;
                 case 2:
-                  cross_product (output_data.boundary_forms[i], data.aux[0][i]);
+                  output_data.boundary_forms[i] = cross_product(data.aux[0][i]);
```
Comment (bangerth) [inline source/fe/mapping_fe_field.cc]: ah, here is where it would break down if you declared the one-argument function only for dim==2. so nevermind.

**Comment by bangerth** [comment]
OK if you eliminate the temporary in wedge product. Please merge yourself.

**Comment by tamiko** [comment]
Thanks.

**Pull request merged**